### ISSUE TITLE
Bug with several GPG recipients

### ISSUE THREAD

**Issue by Christian-Schultz** [dyne/Tomb#337]
I encountering an error when I try to encrypt a key with several GPG keys. 

Consider the following:
`tomb forge secret.key -gr alice,bob`
I have Alice's private key _but only Bob's public key_. This will fail with the message:

> tomb [E] You set an invalid GPG ID.

and that is not what I expect. I am surprised I cannot encrypt with a public key.

I have identified the cause to [line 880 and the following lines ](https://github.com/dyne/Tomb/blob/master/tomb#L880)
```
# At least one private key must be present
for gpg_id in ${recipients[@]}; do
	gpg --list-secret-keys "$gpg_id" &> /dev/null
	[[ $? = 0 ]] && {
		return 0
	}
done
```

This actually checks whether I have the private keys of everyone - contrary to the comment. I expect I only need a single private key - since that would enable me to use the key.

I just want to be sure there is not something I am missing before I submit a pull request. Can you confirm that - like the comment says - it is sufficient with a single private key?

**Comment by roddhjav** [comment]
Hi,

> Consider the following: `tomb forge secret.key -gr alice,bob`

As a first check, you should ensure you have the good key id (always prefer the full fingerprint). You should be able to see alice's key with `gpg -K alice` and bob's key with `gpg -k bob`.

> I have Alice's private key but only Bob's public key.

You need Alice public key too.

> This actually checks whether I have the private keys of everyone - contrary to the comment.

Maybe I missed something but to me it's pretty clear this code return true once it found at least one valid private key.

**Comment by jaromil** [comment]
hi! thanks for looking into this. To me seems there is no bug, but I'm wondering if the case raised is fully covered in tests? another remark is that, at least from a cryptographic point of view, a public key can be always derived from a private key. Not sure if gpg does that implicitly, think not.

**Comment by roddhjav** [comment]
@jaromil This case is tested... but in [pass-tomb](https://github.com/roddhjav/pass-tomb/blob/master/tests/50_errors.sh#L13) where I have the very same function. However, that true, I did not back ported the test in Tomb. I will do it in my next PR.

Regarding GPG, you cannot remove a private key without removing its public key first. However, you might be able to only import a private key. You are right, you can derivative the public key from the private one, but I don't think it is in the GPG spirit to do this kind of think for the user.

**Comment by Christian-Schultz** [comment]
> Hi,
> 
> > Consider the following: `tomb forge secret.key -gr alice,bob`
> 
> As a first check, you should ensure you have the good key id (always prefer the full fingerprint). You should be able to see alice's key with `gpg -K alice` and bob's key with `gpg -k bob`.
> 
> > I have Alice's private key but only Bob's public key.
> 
> You need Alice public key too.
> 
> > This actually checks whether I have the private keys of everyone - contrary to the comment.
> 
> Maybe I missed something but to me it's pretty clear this code return true once it found at least one valid private key.

You are right - the error was in my script. It works as intended, sorry for the confusion. I would like to suggest to raise a warning that no private key is present to give a more clear error message.

**Comment by Narrat** [comment]
Just curious. What kind of error in your script? Misinterpreting the return code?

**Comment by Christian-Schultz** [comment]
> Just curious. What kind of error in your script? Misinterpreting the return code?

No, I simply only tried to use a single GPG user that happened to only have a public key available.
